src/request.zig

```diff
@@ -584,6 +577,69 @@ pub fn parseCookies(self: *const Self, url_encoded: bool) void {
     fio.http_parse_cookies(self.h, if (url_encoded) 1 else 0);
 }
 
+pub const AcceptItem = struct {
+    raw: []const u8,
+    type: Fragment,
+    subtype: Fragment,
+    q: f64,
+
+    const Fragment = union(enum) {
+        glob,
+        value: []const u8,
+    };
+
+    pub fn lessThan(_: void, lhs: AcceptItem, rhs: AcceptItem) bool {
+        return lhs.q < rhs.q;
+    }
+
+    pub fn toContentType(item: AcceptItem) ?ContentType {
+        if (ContentType.string_map.get(item.raw)) |common| {
+            return common;
+        }
+        return null;
+    }
+};
+
+/// Parses `Accept:` http header into `list`, ordered from highest q factor to lowest
+pub fn parseAccept(self: *const Self, list: *std.ArrayList(AcceptItem)) !void {
```

Comment:
Maybe it makes sense to make a type alias for the ArrayList?
So that users can write zap.AcceptList.init() instead of std.ArrayList(zap.AcceptItem).init()?

Reply:
And taking it one step further: instead of passing the ArrayList, let the user just pass an allocator and do the init within parseAccept. I think that would look very zig-like, so I propose:

```zig
pub fn parseAccept(self: *const Self, allocator: std.mem.Allocator) !std.ArrayList(AcceptItem)
```

Reply:
Now, regarding naming:  What do you think of the more verbose `parseAcceptHeaders()`?

```zig
pub fn parseAcceptHeaders(self: *const Self, allocator: std.mem.Allocator) !std.ArrayList(AcceptItem)
```